pkg/manifestclient/write_roundtripper.go

```diff
@@ -144,7 +144,7 @@ func (mrt *writeTrackingRoundTripper) roundTrip(req *http.Request) ([]byte, erro
 	if actionHasRuntimeObjectBody {
 		bodyObj, err = runtime.Decode(unstructured.UnstructuredJSONScheme, bodyContent)
 		if err != nil {
-			return nil, fmt.Errorf("unable to decode body: %w", err)
+			return nil, fmt.Errorf("unable to decode body (content-type is: %s, needed: application/json) : %w", req.Header.Get("Content-Type"), err)
```

Comment:
I would check and fail based on the content-type header before ever trying to decode the body.

Reply:
Updated